### PR TITLE
Support builds-on/runs-on arm

### PR DESCRIPTION
**Pull request by addyess**
Adding a builds-on/runs-on arm64 section for each charm

* Each architecture will have a unique charm for the eventual some-day when we need binary bits in the charms
* Build each architecture with GH builders

### CODE REVIEW

File: charms/worker/charmcraft.yaml [hunk 1]
```diff
@@ -58,6 +72,7 @@ parts:
   charm:
     build-packages: [git]
     charm-entrypoint: k8s/src/charm.py
+    charm-requirements: [k8s/requirements.txt]
```
Comment (addyess) [inline charms/worker/charmcraft.yaml:75]: worker charm can specific it's `requirements.txt` so that the python reqs can be installed with `no-binary`

**Review by mateoflorido — APPROVED** [review]
LGTM

**Comment by addyess** [comment]
@mateoflorido i'm converting this to draft until i get a resolution on:
* https://github.com/canonical/operator-workflows/pull/293

If i have to, we can release from my fork of the publish operator

**Comment by addyess** [comment]
NOW this PR depends on merging of this PR in order to get multiarch builds back
* https://github.com/canonical/operator-workflows/pull/313

File: .github/workflows/integration_test.yaml [hunk 2]
```diff
@@ -24,34 +23,29 @@ jobs:
...
-      extra-arguments: ${{needs.extra-args.outputs.args}} -k test_${{ matrix.suite }}
+      extra-arguments: ${{needs.extra-args.outputs.args}}
       load-test-enabled: false
+      builder-runner-label: ${{matrix.arch.builder-label}}
+      modules: '["test_k8s", "test_etcd"]'
       zap-enabled: false
-      self-hosted-runner: true
-      self-hosted-runner-label: "large"
+      # self-hosted-runner: true
+      # self-hosted-runner-label: ${{matrix.arch.tester-label}}
```
Comment (addyess) [inline .github/workflows/integration_test.yaml:45]: TODO:  re-enable self-hosted runners

File: .github/workflows/integration_test.yaml [hunk 3]
```diff
@@ -24,34 +23,29 @@ jobs:
...
+    uses: addyess/operator-workflows/.github/workflows/integration_test.yaml@multiarch-building
+    needs: [extra-args]
+    secrets: inherit
     strategy:
       matrix:
-        suite: ["k8s", "etcd"]
-    secrets: inherit
+        arch:
+          - {id: amd64, builder-label: ubuntu-22.04, tester-label: large}
+#          - {id: arm64, builder-label: ARM64, tester-label: ARM64}
```
Comment (addyess) [inline .github/workflows/integration_test.yaml:34]: TODO: add arm builders

File: .github/workflows/integration_test.yaml [hunk 4]
```diff
@@ -24,34 +23,29 @@ jobs:
...
-          - "./charms/worker/k8s/"
-          - "./charms/worker/"
-    uses: ./.github/workflows/build-charm.yaml
-    with:
-      working-directory:  ${{ matrix.path }}
-
   integration-tests:
-    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@08c5a65a0bc4696164b4f85a29a9ccbd830d10d8
-    needs: [build-all-charms, extra-args]
+    uses: addyess/operator-workflows/.github/workflows/integration_test.yaml@multiarch-building
```
Comment (addyess) [inline .github/workflows/integration_test.yaml:27]: TODO:  use canonical workflow

**Comment by addyess** [comment]
thanks to @arturo-seijas and @weiiwang01 for unblocking this

File: .github/workflows/integration_test.yaml [hunk 5]
```diff
@@ -41,12 +40,15 @@ jobs:
...
+        - {id: arm64, builder-label: ARM64,        tester-arch: ARM64}
         suite: ["k8s", "etcd", "ceph"]
     with:
       identifier: ${{ matrix.arch.id }}-${{ matrix.suite }}
       builder-runner-label: ${{ matrix.arch.builder-label }}
       charmcraft-channel: ${{ needs.charmcraft-channel.outputs.channel }}
-      extra-arguments: ${{needs.extra-args.outputs.args}} -k test_${{ matrix.suite }}
+      extra-arguments: >-
+        ${{needs.extra-args.outputs.args}} -k test_${{ matrix.suite }}
+        ${{ matrix.arch.id == 'arm64' && ' --lxd-containers' || '' }}
```
Comment (addyess) [inline .github/workflows/integration_test.yaml:51]: I did some study on a failure on ARM now that we're testing with lxd VMs.

lxd is required to use `/dev/kvm`, but on ARM64 -- `/dev/kvm` may not exist.  I even logged into the machine and tried to add `sudo modprobe kvm` but it fails to load the kernel mode.  

without `/dev/kvm` -- lxd doesn't start the machines and the tests fail

Therefore -- this change allows for lxd-containers on ARM -- but i expect we'd need arm64 to support `/dev/kvm` or don't use lxd as a cloud for arm64

**Review by mateoflorido — APPROVED** [review]

**Pull request merged**